### PR TITLE
feat(webapp): create a component chat scaffholding

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest
there is now the following framework.
- a separate page : chat.vue
- a component : Chat.vue
- the chat.vue component is imported into chat.vue
- in the default template page a .chat-modul framework is built in which contains the Chat.vue component but with the prop singleRoom = true  
- a button in the userprofile i visit. to show the single chat module. 

this ensures to chat with a user without leaving the visited page. 

this is pulled from the branch  #6398  

https://github.com/Ocelot-Social-Community/Ocelot-Social/assets/1324583/9e22d477-a276-4aff-9169-da3a5324cec1

based on: https://github.com/Ocelot-Social-Community/Ocelot-Social/pull/6505

### CODE REVIEW

File: webapp/components/utils/InternalPages.js [hunk 1]
```diff
@@ -1,6 +1,27 @@
...
+      footerIdent: 'chatX', // localized string identifier
+      headTitleIdent: 'chatX', // localized string identifier
+      headlineIdent: 'chatX', // localized string identifier. on null it's hidden, on empty string default is used
+      hasContainer: true,
+      hasBaseCard: true,
+      hasLoginInHeader: true,
+      // in case internal page content is here 'webapp/locales/html/'
+      // htmlIdent: 'html.chat',
+    },
+  }),
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/utils/InternalPages.js]: done

File: webapp/pages/chat.vue [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<template>
+  <div>
+    <chat :theme="theme" />
+    <base-button @click="theme = theme === 'light' ? 'dark' : 'light'">
+      change style mode
+    </base-button>
+  </div>
+</template>
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/pages/chat.vue]: done

File: webapp/pages/chat.vue [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    return {
+      theme: 'light',
+    }
+  },
+}
+</script>
+<style lang="scss">
+body {
+  font-family: 'Quicksand', sans-serif;
+}
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/pages/chat.vue]: done

File: webapp/store/chat-modul.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+export const state = () => {
+  return {
+    showChatModul: false,
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/store/chat-modul.js]: done

File: webapp/constants/links.js [hunk 5]
```diff
@@ -140,6 +140,7 @@ export default {
 
   FOOTER_LINK_LIST: [
     ORGANIZATION,
+
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/constants/links.js:143]: done

File: webapp/components/Chat/Chat.vue [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+          title: 'Invite User',
+        },
+        {
+          name: 'removeUser',
+          title: 'Remove User',
+        },
+        {
+          name: 'deleteRoom',
+          title: 'Delete Room',
+        },
```
Comment (Elweyn) [inline webapp/components/Chat/Chat.vue:59]: I think that we don't need this
```suggestion
```

File: webapp/components/Chat/Chat.vue [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+      ],
+      messageActions: [
+        {
+          name: 'addMessageToFavorite',
+          title: 'Add To Favorite',
+        },
+        {
+          name: 'shareMessage',
+          title: 'Share Message',
+        },
```
Comment (Elweyn) [inline webapp/components/Chat/Chat.vue:69]: Do we want this?
If not:
```suggestion
```

File: webapp/components/Chat/Chat.vue [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+      ],
+      templatesText: [
+        {
+          tag: 'help',
+          text: 'This is the help',
+        },
+        {
+          tag: 'action',
+          text: 'This is the action',
+        },
```
Comment (Elweyn) [inline webapp/components/Chat/Chat.vue:79]: Do we need this?
If not:
```suggestion
```

File: webapp/components/Chat/Chat.vue [hunk 9]
```diff
@@ -0,0 +1,241 @@
...
+        NEW_MESSAGES: 'Nouveaux messages',
+        MESSAGE_DELETED: 'Ce message a été supprimé',
+        MESSAGES_EMPTY: 'Aucun message',
+        CONVERSATION_STARTED: 'La conversation a commencée le :',
+        TYPE_MESSAGE: 'Tapez votre message',
+        SEARCH: 'Rechercher',
+        IS_ONLINE: 'est en ligne',
+        LAST_SEEN: 'dernière connexion ',
+        IS_TYPING: 'est en train de taper...',
+        CANCEL_SELECT_MESSAGE: 'Annuler Sélection',
```
Comment (Elweyn) [inline webapp/components/Chat/Chat.vue:93]: Need to be internationalized

File: webapp/components/Chat/Chat.vue [hunk 10]
```diff
@@ -0,0 +1,241 @@
...
+        {
+          roomId: '2',
+          roomName: 'Max J. Mustermann',
+          avatar:
+            'https://64.media.tumblr.com/8889b6e26370f4e3837584c1c59721a6/f4f76ed6b0249d08-4b/s1280x1920/810e9e5fa724366d26c10c0fa22ba97dad8778d1.pnj',
+          users: [
+            { _id: '1234', username: 'Johnx Doe' },
+            { _id: '43210', username: 'Max J. Mustermann' },
+          ],
+        },
```
Comment (Elweyn) [inline webapp/components/Chat/Chat.vue:123]: Need to be queried

File: webapp/components/Chat/Chat.vue [hunk 11]
```diff
@@ -0,0 +1,241 @@
...
+            // console.log('Messages', Message)
+            this.messages = Message
+          } catch (error) {
+            // console.log('Error', error)
+            this.messages = [] // this.addMessages(true)
+            this.$toast.error(error.message)
+          }
+        }
+        this.messagesLoaded = true
+      })
```
Comment (Elweyn) [inline webapp/components/Chat/Chat.vue:159]: Need to get a subscription

File: webapp/components/ChatNotificationMenu/ChatNotificationMenu.vue [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+    Dropdown,
+    // NotificationList,
+  },
+  data() {
+    return {
+      notifications: [
+        { roomid: 'u1', name: 'Jenny', title: 'last Message from Jenny' },
+        { roomid: 'u2', name: 'Honey', title: 'last Message from Honey' },
+        { roomid: 'u3', name: 'Bob der Baumeister', title: 'last Message from Bob der Baumeister' },
+      ],
```
Comment (Elweyn) [inline webapp/components/ChatNotificationMenu/ChatNotificationMenu.vue:79]: Need to be queried

**Review by Elweyn — APPROVED** [review]

File: webapp/components/Chat/Chat.vue (hunk 6 above)
Comment (Elweyn) [reply webapp/components/Chat/Chat.vue:59]: Will done in PR from @maeckes-infinitylabs

File: webapp/components/Chat/Chat.vue (hunk 7 above)
Comment (Elweyn) [reply webapp/components/Chat/Chat.vue:69]: Will be done by @maeckes-infinitylabs?

File: webapp/components/Chat/Chat.vue (hunk 8 above)
Comment (Elweyn) [reply webapp/components/Chat/Chat.vue:79]: Will be done by @maeckes-infinitylabs?

File: webapp/components/Chat/Chat.vue (hunk 9 above)
Comment (Elweyn) [reply webapp/components/Chat/Chat.vue:93]: Will be done by @maeckes-infinitylabs?

File: webapp/components/Chat/Chat.vue (hunk 11 above)
Comment (Elweyn) [reply webapp/components/Chat/Chat.vue:159]: https://github.com/Ocelot-Social-Community/Ocelot-Social/issues/6502

File: webapp/layouts/default.spec.js [hunk 13]
```diff
@@ -36,6 +36,9 @@ describe('default.vue', () => {
     store = new Vuex.Store({
       getters: {
         'auth/isLoggedIn': () => true,
+        'chat/showChat': () => {
+          return { showChat: false, roomID: 'u0' }
+        },
```
[the remark this answers is not in the document]
Comment (Elweyn) [reply webapp/layouts/default.spec.js:41]: Done in new PR from @ogerly

File: webapp/components/Chat/Chat.vue (hunk 10 above)
Comment (Elweyn) [reply webapp/components/Chat/Chat.vue:123]: #6506

File: webapp/components/Chat/Chat.vue [hunk 14]
```diff
@@ -0,0 +1,241 @@
...
+      this.messages = [
+        ...this.messages,
+        {
+          _id: this.messages.length,
+          content: message.content,
+          senderId: this.currentUserId,
+          timestamp: new Date().toString().substring(16, 21),
+          date: new Date().toDateString(),
+        },
+      ]
```
[the remark this answers is not in the document]
Comment (Elweyn) [reply webapp/components/Chat/Chat.vue:197]: #6506

File: webapp/components/ChatNotificationMenu/ChatNotificationMenu.vue (hunk 12 above)
Comment (Elweyn) [reply webapp/components/ChatNotificationMenu/ChatNotificationMenu.vue:79]: #6506

File: webapp/components/Chat/Chat.vue (hunk 7 above)
Comment (ogerly) [reply webapp/components/Chat/Chat.vue:69]: placeholder

File: webapp/components/Chat/Chat.vue (hunk 9 above)
Comment (ogerly) [reply webapp/components/Chat/Chat.vue:93]: in  work from @maeckes-infinitylabs

File: webapp/components/Chat/Chat.vue [hunk 15]
```diff
@@ -0,0 +1,241 @@
...
+          ...this.messages,
+          {
+            _id: this.messages.length,
+            content: 'NEW MESSAGE',
+            senderId: '1234',
+            timestamp: new Date().toString().substring(16, 21),
+            date: new Date().toDateString(),
+          },
+        ]
+      }, 2000)
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Chat/Chat.vue:212]: 1. send = send
2. add a message in chat timeline

**Pull request merged**